besu/src/main/java/org/hyperledger/besu/controller/MergeBesuControllerBuilder.java

```diff
@@ -186,19 +187,24 @@ protected MergeContext createConsensusContext(
       final WorldStateArchive worldStateArchive,
       final ProtocolSchedule protocolSchedule) {
 
-    OptionalLong terminalBlockNumber = configOptionsSupplier.get().getTerminalBlockNumber();
-    Optional<Hash> terminalBlockHash = configOptionsSupplier.get().getTerminalBlockHash();
+    final GenesisConfigOptions genesisConfigOptions = configOptionsSupplier.get();
+    final OptionalLong terminalBlockNumber = genesisConfigOptions.getTerminalBlockNumber();
+    final Optional<Hash> terminalBlockHash = genesisConfigOptions.getTerminalBlockHash();
+    final boolean isPostMergeAtGenesis =
+        genesisConfigOptions.getTerminalTotalDifficulty().isPresent()
+            && genesisConfigOptions.getTerminalTotalDifficulty().get().isZero()
+            && blockchain.getGenesisBlockHeader().getDifficulty().isZero();
```

Comment:
If everything is the same except difficulty is 1, does it still work?

Reply:
discussed in person: TTD and D = 0 is on-spec for postMergeGenesis. If anyone created a new network with TTD = 0 and D = 1 then that's a misconfiguration.